lib/metrics/AmqpMetricReporter.ts

```diff
@@ -39,9 +52,51 @@ export interface AmqpMetricReporterOptions extends ScheduledMetricReporterOption
   connection: string;
   exchangeName: string;
   queueName: string;
+
+  /**
+   * Used to build the amqp message for a metric.
+   * @type {MetricMessageBuilder}
+   */
+  metricMessageBuilder: MetricMessageBuilder;
 }
 
-export class AmqpMetricReporter extends ScheduledMetricReporter<AmqpMetricReporterOptions, {}> {
+export class AmqpMetricReporter extends ScheduledMetricReporter<AmqpMetricReporterOptions, Amqp.Message> {
+  /**
+   * Returns a {@link MetricMessageBuilder} that builds an Amqp.Message for a metric.
+   *
+   * @static
+   * @returns {MetricMessageBuilder}
+   * @memberof AmqpMetricReporter
+   */
+  public static defaultMessageBuilder(): MetricMessageBuilder {
+    return (registry: MetricRegistry, metric: Metric, type: MetricType, timestamp: Date, tags: Tags) => {
+      let values = null;
+
+      if (metric instanceof MonotoneCounter) {
+        values = AmqpMetricReporter.getMonotoneCounterValues(metric);
+      } else if (metric instanceof Counter) {
+        values = AmqpMetricReporter.getCounterValues(metric);
+      } else if (metric instanceof Histogram) {
+        values = AmqpMetricReporter.getHistogramValues(metric);
+      } else if (metric instanceof Meter) {
+        values = AmqpMetricReporter.getMeterValues(metric);
+      } else if (metric instanceof Timer) {
+        values = AmqpMetricReporter.getTimerValues(metric);
+      } else {
+        values = AmqpMetricReporter.getGaugeValue(metric as Gauge<any>);
+      }
+
+      if (values === null) {
```

Comment:
This should do the job as well
```suggestion
      if (!values) {
```